includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -220,7 +220,7 @@ public function get_product_price( $product ) {
 		}
 
 		// Add subscription sign-up fees to product price.
-		if ( 'subscription' === $product->get_type() && class_exists( 'WC_Subscriptions_Product' ) ) {
+		if ( 'subscription' === $product->get_type() && class_exists( 'WC_Subscriptions_Product' ) && is_numeric( $product_price ) ) {
```

Comment:
We should return a default value for `$product_price` if both the conditions in this function are false.

Reply:
`get_price()` function is something to be careful with, it has different values for different cases, see https://github.com/Automattic/woocommerce-payments/pull/1774#discussion_r634867263 @jessy-p is creating an issue to improve documentation around it.

Reply:
FYI: If the product has no price, it returns an empty string and an integer if it has a value.